jupyterhub/auth.py

```diff
@@ -762,7 +771,15 @@ def add_system_user(self, user):
         Tested to work on FreeBSD and Linux, at least.
         """
         name = user.name
-        cmd = [arg.replace('USERNAME', name) for arg in self.add_user_cmd] + [name]
+        cmd = [arg.replace('USERNAME', name) for arg in self.add_user_cmd]
+        try:
+            uid = self.uids[name]
+            cmd += ['--uid', '%d' % uid]
+        except AttributeError:
```

Comment:
@minrk Currently the `uids` attribute is totally optional (since I made it up in this PR).  I think AttributeError would only be caught if the user just hadn't provided it.  Or would you prefer an explicit check for that attribute first, instead of catching the exception?

Reply:
It's not optional. Since it's defined in a traitlet above, it is guaranteed to exist and be a dict. The default is an empty dict.

Reply:
Oh, right.  I've removed the AttributeError catch.